### PR TITLE
Migrate to 3.x

### PR DESCRIPTION
**Pull request by jjnesbitt**
Draft because the `girder build` command is failing. The error it produces is:

```
ERROR in plugin.min.js from UglifyJs
Unexpected token: operator (-) [plugin.min.js:1,22]
```

The direct command that causes this is:

`grunt "--static-path=/home/jacob/.virtualenvs/girder/share/girder/static" "--static-public-path=/static" "--no-progress=false" "--env=prod"`

Besides this, the plugin seems to load into girder just fine, appearing in the plugins tab.

### CODE REVIEW

File: tape_archive/web_client/package.json
Comment (brianhelba) [inline tape_archive/web_client/package.json]: The hyphen in this name seems to be causing the problem.

For reference, the issue is that this name included in [Webpack's `output.library` property](https://webpack.js.org/configuration/output/#outputlibrary), which is configured to output the first line of the [built plugin file as `var girder_plugin_${name} =`](https://github.com/girder/girder/blob/460f598604362312ab715b64e628e2f99cffee6c/girder/web_client/grunt_tasks/build.js#L226). After this file is emitted (internally) the Uglify tool comes around to minify it, and sees `var girder_plugin_tape-archive =`, which is invalid, hence the error.

Following the practices of other Girder plugins with compound names, I think the convention seems to be to use underscore names for almost everything, except the name of the Pip package that gets published. So, the `name=` field of `setup.py` should use `tape-archive`, and everything else (variables and directory names) should use `tape_archive`.

File: tape_archive/web_client/main.js
Comment (brianhelba) [inline tape_archive/web_client/main.js]: I think you should add `import $ from 'jquery';` here too, for correctness.

File: tape_archive/web_client/TarExportView.js
Comment (brianhelba) [inline tape_archive/web_client/TarExportView.js]: I think you should add `import $ from 'jquery';` here too, for correctness.

File: setup.py
Comment (brianhelba) [inline setup.py]: Use [String literal concatenation](https://docs.python.org/2.0/ref/string-catenation.html) to split this over multiple lines. Otherwise, we'll end up with whitespace in the string.

File: setup.py
Comment (brianhelba) [inline setup.py]: Extra newlines.

File: setup.py
Comment (brianhelba) [inline setup.py]: Is this import needed?

File: tape_archive/__init__.py
Comment (brianhelba) [inline tape_archive/__init__.py]: Remove before finalizing.

File: tape_archive/__init__.py
Comment (brianhelba) [inline tape_archive/__init__.py]: Remove before finalizing.

**Review by brianhelba — COMMENTED** [review]
Some feedback on how to fix build issues.

**Comment by jjnesbitt** [comment]
Just adding a screenshot to show the dashboard that renders dashboard

**Review by brianhelba — APPROVED** [review]
LGTM.

**Comment by brianhelba** [comment]
@zachmullen We've verified that this successfully loads, but have not done any testing of the functionality. I'll let you do additional testing and review as you see fit. Feel free to merge this when you're satisfied.

**Comment by brianhelba** [comment]
@zachmullen Also, publishing is now set up, but we haven't run it yet. When this is merged to master, you should initiate the first publish yourself.

**Comment by brianhelba** [comment]
`tox -e release`

**Review by zachmullen — APPROVED** [review]

**Pull request merged**
